### PR TITLE
#112 - bugfix to work as intended.

### PR DESCRIPTION
**Pull request by keydam**
It now works as intended, can you check ?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 0.6.1
+
+* #112 - bugfix to work as intended.
```
Comment (swatson31) [inline CHANGELOG.md:3]: I suggest describing what was fixed. This comment isn't very helpful without digging into the related code changes.

File: src/main/java/com/github/johrstrom/listener/PrometheusServer.java [hunk 2]
```diff
@@ -161,8 +162,9 @@ public synchronized void start() throws IOException {
         server.start();      
     }
     
-    public synchronized void stop() {
-    	server.stop(delay);
+    public synchronized void stop() throws InterruptedException {
+        TimeUnit.SECONDS.sleep(delay);
+    	server.stop(0);
```
Comment (swatson31) [inline src/main/java/com/github/johrstrom/listener/PrometheusServer.java:167]: I haven't looked deeper, but I wonder if the problem should be fixed inside of the stop() method. It looks like non-zero stop() is broken by the way you fixed this. I think you should at least leave a comment explaining why fixing the server.stop() method is not preferred.

**Review by swatson31 — CHANGES_REQUESTED** [review]
I'm not an official maintainer of this repo, but I found it very useful at work. I care about the future of this code.  I have a few concerns about this change which are called out at the specific line numbers...

File: src/main/java/com/github/johrstrom/listener/PrometheusServer.java (hunk 2 above)
Comment (swatson31) [reply src/main/java/com/github/johrstrom/listener/PrometheusServer.java:167]: I found the issue you created and read [the doc you linked](https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpServer.html#stop-int-). I would support this change with at least a more descriptive change in the CHANGELOG.  👍

**Comment by johrstrom** [comment]
Hi, thanks for the pull request. I'll try to look into this weekend. I lost the gpg key I used to sign these jars so I've got to figure some way to release.

That said - on the surface it looks OK, but I'll have to check the docs on `stop` a little.

**Comment by keydam** [comment]
Any news ?
I can indeed be more specific in the changelog if necessary

**Comment by johrstrom** [comment]
Hi, sorry for the delay. No, there's nothing more on your side I require I just need to find the time is all. I can't work on this at work, so I need to free up some personal time to get it all sorted & released.

**Comment by johrstrom** [comment]
I was going to pull this in, but it doesn't look like the commit you made is actually tied to your github user.  Meaning, you're github account https://github.com/keydam is not going to be credited with the commit (an unknown user is).

You can fix the commit if you like. I'd suggest you do just so you get the proper credit.

Aside from that - I'm working on automation to make releases. As I've said, I've lost the gpg key I used to sign these to upload to maven, so 0.6.1 (the next version) may just be attached to this repo and not on maven.  That siad, I'll still work to actually upload them to maven central as well.

**Pull request closed without merging**